platform/pulpcore/tasking/tasks.py

```diff
@@ -243,13 +243,22 @@ def apply_async(self, tags=[], group_id=None, args=None, kwargs=None, **options)
 
     def __call__(self, *args, **kwargs):
         """
-        Set the :class:`pulpcore.app.models.Task` object in the running state and log some output.
+        Set the :class:`pulpcore.app.models.Task` object in the running state, set the worker,
+        and log some output.
 
         Skip the status updating if the task is called synchronously.
         """
-        if not self.request.called_directly:
-            task_status = TaskStatus.objects.get(pk=self.request.id)
-            task_status.set_running()
+        task_status = TaskStatus.objects.get(pk=self.request.id)
+        task_status.set_running()
+
+        worker = None
+        try:
+            worker = Worker.objects.get(name=self.request.hostname)
+        except Worker.DoesNotExist:
+            _logger.warn("Failed to find worker with hostname {host}".format(
+                host=self.request.hostname))
+
+        task_status.set_worker(worker)
```

Comment:
This only needs to be set if L#256 does not raise an exception. In that case, L#254 can be deleted and this line L#261 can be moved to an `else:` block as part of the `try/except` block.